sopel/modules/admin.py

```diff
@@ -247,61 +247,85 @@ def mode(bot, trigger):
     bot.write(('MODE', bot.nick + ' ' + mode))
 
 
-@sopel.module.require_privmsg("This command only works as a private message.")
-@sopel.module.require_admin("This command requires admin privileges.")
-@sopel.module.commands('set')
-@sopel.module.example('.set core.owner Me')
-def set_config(bot, trigger):
-    """See and modify values of Sopel's config object.
-
-    Trigger args:
-        arg1 - section and option, in the form "section.option"
-        arg2 - value
-
-    If there is no section, section will default to "core".
-    If value is None, the option will be deleted.
+def parse_section_option_value(bot, trigger):
+    """Parse trigger for set/unset to get relevant config elements.
+
+    Returns:
+        - False, if invalid command format; let command handle sending usage.
```

Comment:
I see where you are coming from: language like C or Go uses `if err == -1` or alike as their control-flow mechanism—which is fine for these languages. The pythonic way for control flow is to use exceptions instead.

For example, instead of `return False` you can use `raise ValueError`.

Then, in both `set_config` and `unset_config`, you can do that:

```
try:
    section, section_name, static_sec, option, value = parse_section_option_value(
        bot.config, trigger)
except ValueError:
    bot.reply('Usage: .set section.option [value]')
    return
except InvalidSection as error:
    bot.reply("Section %s does not exist." % (error.section))
    return
except InvalidOption as error:
    bot.reply("Option %s.%s does not exist." % (error.section, error.option))
    return
```

Provided you have `InvalidSection` and `InvalidOption ` exception class at your disposal—if not, create them!

Also, notice how I replace the parameter `bot` by `bot.config` in `parse_section_option_value`.

Reply:
Nice! I like that a lot better. I'm happy to be more idiomatic :grin: 

Suggested changes have been made.